### PR TITLE
Fix transport endpoint issue for rclone mounter

### PR DESCRIPTION
**Pull request by Bhagyashreek8**

### CODE REVIEW

**Comment by Bhagyashreek8** [comment]
Test results added in GHE - https://github.ibm.com/alchemy-containers/armada-storage/issues/5726#issuecomment-76658239

**Comment by Bhagyashreek8** [comment]
UT test results
```
$ go test -v -coverpkg=./... -coverprofile=profile.cov ./... && cat profile.cov | grep -v /controllers/fake/ | grep -v /api/ > profile2.cov && go tool cover -func profile2.cov
?   	github.com/IBM/ibm-object-csi-driver-operator	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/api/v1alpha1	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_create	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_delete	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_get	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_get/csidriver	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_get/pvc	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_get/serviceaccount	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_get/deployment	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_list	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_list/nodeserverpod	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_update	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/fake/client_update/ibmobjectcsi	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/common	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/util	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/pkg/config	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/pkg/util/boolptr	[no test files]
?   	github.com/IBM/ibm-object-csi-driver-operator/version	[no test files]
=== RUN   TestIBMObjectCSIReconcile
=== RUN   TestIBMObjectCSIReconcile/Positive:_Successful
=== RUN   TestIBMObjectCSIReconcile/Positive:_IBMObjectCSI_CR_not_found
=== RUN   TestIBMObjectCSIReconcile/Positive:_Sync_controller_deployment_&_pod_containers_and_update_status_in_IBMObjectCSI_CR
=== RUN   TestIBMObjectCSIReconcile/Positive:_Successfully_updated_status_in_IBMObjectCSI_CR_after_validating_if_pod_images_are_in_sync
=== RUN   TestIBMObjectCSIReconcile/Positive:_Successfully_removed_finaliser_from_IBMObjectCSI_CR
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_get_IBMObjectCSI_CR
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_Add_Finalizer_in_IBMObjectCSI_CR
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_create_CSI_driver_while_reconciling
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_get_CSI_driver_while_reconciling
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_create_service_account_while_reconciling
=== RUN   TestIBMObjectCSIReconcile/Failed_to_get_node_daemon_set_while_reconciling
=== RUN   TestIBMObjectCSIReconcile/Failed_to_get_controller_deployment_while_reconciling
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_restart_node_while_reconciling
=== RUN   TestIBMObjectCSIReconcile/Failed_to_get_service_account_while_reconciling
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_get_controller_pod_while_reconciling
=== RUN   TestIBMObjectCSIReconcile/Negative:_Controller_pod_not_found_while_reconciling
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_sync_CSI_Controller
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_sync_CSI_Node
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_update_status_in_IBMObjectCSI_CR
=== RUN   TestIBMObjectCSIReconcile/Negative:_IBMObjectCSI_CR_is_deleted_and_failed_to_delete_cluster_role_binding
=== RUN   TestIBMObjectCSIReconcile/Negative:_IBMObjectCSI_CR_is_deleted_and_failed_to_delete_cluster_role
=== RUN   TestIBMObjectCSIReconcile/Negative:_IBMObjectCSI_CR_is_deleted_and_failed_to_delete_storage_class
=== RUN   TestIBMObjectCSIReconcile/Negative:_IBMObjectCSI_CR_is_deleted_and_failed_to_delete_CSI_driver
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_get_CSI_driver_while_deleting
=== RUN   TestIBMObjectCSIReconcile/Negative:_Failed_to_remove_finaliser_from_IBMObjectCSI_CR
--- PASS: TestIBMObjectCSIReconcile (0.02s)
    --- PASS: TestIBMObjectCSIReconcile/Positive:_Successful (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Positive:_IBMObjectCSI_CR_not_found (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Positive:_Sync_controller_deployment_&_pod_containers_and_update_status_in_IBMObjectCSI_CR (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Positive:_Successfully_updated_status_in_IBMObjectCSI_CR_after_validating_if_pod_images_are_in_sync (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Positive:_Successfully_removed_finaliser_from_IBMObjectCSI_CR (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_get_IBMObjectCSI_CR (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_Add_Finalizer_in_IBMObjectCSI_CR (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_create_CSI_driver_while_reconciling (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_get_CSI_driver_while_reconciling (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_create_service_account_while_reconciling (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Failed_to_get_node_daemon_set_while_reconciling (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Failed_to_get_controller_deployment_while_reconciling (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_restart_node_while_reconciling (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Failed_to_get_service_account_while_reconciling (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_get_controller_pod_while_reconciling (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Controller_pod_not_found_while_reconciling (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_sync_CSI_Controller (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_sync_CSI_Node (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_update_status_in_IBMObjectCSI_CR (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_IBMObjectCSI_CR_is_deleted_and_failed_to_delete_cluster_role_binding (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_IBMObjectCSI_CR_is_deleted_and_failed_to_delete_cluster_role (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_IBMObjectCSI_CR_is_deleted_and_failed_to_delete_storage_class (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_IBMObjectCSI_CR_is_deleted_and_failed_to_delete_CSI_driver (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_get_CSI_driver_while_deleting (0.00s)
    --- PASS: TestIBMObjectCSIReconcile/Negative:_Failed_to_remove_finaliser_from_IBMObjectCSI_CR (0.00s)
=== RUN   TestIBMObjectCSISetupWithManager
=== RUN   TestIBMObjectCSISetupWithManager/Positive:_Successful
--- PASS: TestIBMObjectCSISetupWithManager (0.00s)
    --- PASS: TestIBMObjectCSISetupWithManager/Positive:_Successful (0.00s)
=== RUN   TestRecoverStaleVolumeReconcile
=== RUN   TestRecoverStaleVolumeReconcile/Positive:_Successful
=== RUN   TestRecoverStaleVolumeReconcile/Positive:_RecoverStaleVolume_CR_not_found
=== RUN   TestRecoverStaleVolumeReconcile/Positive:_No_workload_is_passed_to_watch
=== RUN   TestRecoverStaleVolumeReconcile/Positive:_Node_Server_Pods_not_found
=== RUN   TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_RecoverStaleVolume_CR
=== RUN   TestRecoverStaleVolumeReconcile/Negative:_Deployment_Name_is_missing
=== RUN   TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_deployment
=== RUN   TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_deployment_pods_list
=== RUN   TestRecoverStaleVolumeReconcile/Megative:_Failed_to_get_pvc
=== RUN   TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_node-server-pods_list
=== RUN   TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_In_Cluster_Config
=== RUN   TestRecoverStaleVolumeReconcile/Negative:_Failed_to_delete_pod
--- PASS: TestRecoverStaleVolumeReconcile (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Positive:_Successful (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Positive:_RecoverStaleVolume_CR_not_found (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Positive:_No_workload_is_passed_to_watch (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Positive:_Node_Server_Pods_not_found (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_RecoverStaleVolume_CR (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Negative:_Deployment_Name_is_missing (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_deployment (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_deployment_pods_list (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Megative:_Failed_to_get_pvc (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_node-server-pods_list (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Negative:_Failed_to_get_In_Cluster_Config (0.00s)
    --- PASS: TestRecoverStaleVolumeReconcile/Negative:_Failed_to_delete_pod (0.00s)
=== RUN   TestRecoverStaleVolumeSetupWithManager
=== RUN   TestRecoverStaleVolumeSetupWithManager/Positive:_Successful
--- PASS: TestRecoverStaleVolumeSetupWithManager (0.00s)
    --- PASS: TestRecoverStaleVolumeSetupWithManager/Positive:_Successful (0.00s)
=== RUN   TestContains
=== RUN   TestContains/Positive:_Successful
--- PASS: TestContains (0.00s)
    --- PASS: TestContains/Positive:_Successful (0.00s)
=== RUN   TestCreateK8sClient
=== RUN   TestCreateK8sClient/#00
--- PASS: TestCreateK8sClient (0.00s)
    --- PASS: TestCreateK8sClient/#00 (0.00s)
=== RUN   TestAPIs
Running Suite: Controller Suite - /Users/bhagyashree/Documents/ObjectStorage/csi-driver/cis-operator/ibm-object-csi-driver-operator/controllers
===============================================================================================================================================
Random Seed: 1712580865

Will run 0 of 0 specs

Ran 0 of 0 Specs in 0.000 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestAPIs (0.00s)
PASS
coverage: 23.6% of statements in ./...
ok  	github.com/IBM/ibm-object-csi-driver-operator/controllers	1.449s	coverage: 23.6% of statements in ./...
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:99:			Reconcile					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:180:			updateStatus					85.7%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:227:			restartControllerPodfromDeployment		100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:236:			getControllerPod				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:254:			areAllPodImagesSynced				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:274:			isControllerReady				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:278:			isNodeReady					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:282:			getNodeDaemonSet				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:291:			getControllerDeployment				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:300:			reconcileClusterRoleBinding			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:305:			reconcileStorageClasses				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:310:			reconcileClusterRole				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:315:			reconcileServiceAccount				97.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:375:			rolloutRestartNode				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:382:			restartControllerPod				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:405:			reconcileCSIDriver				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:428:			deleteCSIDriver					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:452:			deleteClusterRoleBindings			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:457:			deleteStorageClasses				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:462:			getClusterRoleBindings				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:474:			getStorageClasses				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:484:			deleteClusterRoles				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:489:			getClusterRoles					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/ibmobjectcsi_controller.go:502:			SetupWithManager				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/common/common.go:5:				GetSelectorLabels				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/common.go:14:			GetImagePullSecrets				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:19:			New						100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:26:			Unwrap						100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:31:			GetLabels					50.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:53:			GetAnnotations					50.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:71:			GetCSINodeSelectorLabels			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:75:			GetCSINodePodLabels				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:79:			GetCSINodeImage					66.7%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:86:			GetCSIControllerSelectorLabels			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:90:			GetCSIControllerPodLabels			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:94:			GetCSIControllerImage				66.7%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:101:			GetCSIControllerResourceRequests		100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:127:			GetCSINodeResourceRequests			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/ibmobjectcsi.go:153:			GetCSIResource					0.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:12:	GenerateCSIDriver				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:27:	GenerateControllerServiceAccount		100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:31:	GenerateNodeServiceAccount			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:35:	getServiceAccount				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:47:	GenerateExternalProvisionerClusterRole		100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:92:	GenerateExternalProvisionerClusterRoleBinding	100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:112:	GenerateSCCForControllerClusterRole		100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:128:	GenerateSCCForControllerClusterRoleBinding	100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:148:	GenerateSCCForNodeClusterRole			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:169:	GenerateSCCForNodeClusterRoleBinding		100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:188:	Generates3fsSC					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/internal/crutils/static_resource_generator.go:215:	GenerateRcloneSC				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/logParser.go:15:					getLogEntry					88.9%
github.com/IBM/ibm-object-csi-driver-operator/controllers/logParser.go:37:					parseLogs					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/recoverstalevolume_controller.go:72:			Reconcile					98.3%
github.com/IBM/ibm-object-csi-driver-operator/controllers/recoverstalevolume_controller.go:276:			SetupWithManager				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/recoverstalevolume_controller.go:282:			contains					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/recoverstalevolume_controller.go:291:			createK8sClient					42.9%
github.com/IBM/ibm-object-csi-driver-operator/controllers/recoverstalevolume_controller.go:307:			fetchVolumeStatsFromNodeServerLogs		88.2%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/common.go:14:					ensureAnnotations				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:38:				NewCSIControllerSyncer				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:68:				SyncFn						90.9%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:91:				ensurePodSpec					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:105:				ensureContainersSpec				94.4%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:168:				ensureDefaultResources				0.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:172:				ensureResources					0.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:187:				ensureContainer					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:202:				getEnvFor					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:224:				getVolumeMountsFor				75.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:245:				ensureVolumes					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:253:				getSidecarByName				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:257:				getSidecarImageByName				75.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:265:				getCSIProvisionerImage				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:269:				getLivenessProbeImage				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:273:				getSidecarPullPolicy				75.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:281:				getCSIProvisionerPullPolicy			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:285:				getLivenessProbePullPolicy			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:289:				ensurePorts					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:293:				ensureProbe					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:304:				ensureVolume					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:311:				getSidecarByName				75.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:320:				getCSIControllerResourceRequests		100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_controller.go:339:				getSidecarResourceRequests			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:41:				NewCSINodeSyncer				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:71:				SyncFn						90.9%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:93:				ensurePodSpec					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:101:				ensureContainersSpec				95.7%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:181:				ensureContainer					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:192:				envVarFromField					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:205:				getEnvFor					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:232:				getVolumeMountsFor				83.3%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:289:				ensureVolumes					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:301:				getSidecarByName				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:305:				getCSINodeDriverRegistrarImage			75.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:313:				getLivenessProbeImage				75.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:321:				getCSINodeDriverRegistrarPullPolicy		75.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:329:				ensureHostPathVolumeSource			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:340:				fillSecurityContextCapabilities			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/syncer/csi_node.go:354:				getCSINodeResourceRequests			100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/test_variables.go:49:					setupScheme					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:27:				NewControllerHelper				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:33:				DeleteClusterRoleBindings			84.6%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:53:				DeleteStorageClasses				84.6%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:73:				ReconcileClusterRoleBinding			69.2%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:92:				ReconcileStorageClasses				69.2%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:111:				getClusterRoleBinding				100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:120:				getStorageClass					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:129:				DeleteClusterRoles				84.6%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:149:				ReconcileClusterRole				50.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:173:				getClusterRole					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:182:				AddFinalizerIfNotPresent			90.9%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:203:				RemoveFinalizer					88.9%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/common/common.go:220:				getAccessorAndFinalizerName			63.6%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/datastructures.go:3:				Contains					100.0%
github.com/IBM/ibm-object-csi-driver-operator/controllers/util/datastructures.go:12:				Remove						80.0%
github.com/IBM/ibm-object-csi-driver-operator/pkg/config/resources.go:10:					String						100.0%
github.com/IBM/ibm-object-csi-driver-operator/pkg/config/resources.go:30:					GetNameForResource				100.0%
github.com/IBM/ibm-object-csi-driver-operator/pkg/util/boolptr/boolptr.go:4:					True						100.0%
github.com/IBM/ibm-object-csi-driver-operator/pkg/util/boolptr/boolptr.go:10:					False						100.0%
total:														(statements)					90.6%
Bhagyashrees-MacBook-Pro-2:ibm-object-csi-driver-operator bhagyashree$ 
```

File: api/v1alpha1/ibmobjectcsi_types.go [hunk 1]
```diff
@@ -36,6 +36,18 @@ type IBMObjectCSISpec struct {
 	ImagePullSecrets []string `json:"imagePullSecrets,omitempty"`
 
 	HealthPort uint16 `json:"healthPort,omitempty"`
+
+	Resources ResourcesSpec `json:"resources,omitempty"`
```
Comment (ashimagarg27) [inline api/v1alpha1/ibmobjectcsi_types.go:40]: Why we need resources at IBMObject CSI CR level?

File: controllers/internal/crutils/ibmobjectcsi.go [hunk 2]
```diff
@@ -97,3 +97,28 @@ func (c *IBMObjectCSI) GetCSIControllerImage() string {
...
+	}
+	return &resources
+}
+
+func (c *IBMObjectCSI) GetCSIResource(image string) string {
+	if c.Spec.Controller.Tag == "" {
+		return c.Spec.Controller.Repository
+	}
+	return c.Spec.Controller.Repository + ":" + c.Spec.Controller.Tag
+}
```
Comment (ashimagarg27) [inline controllers/internal/crutils/ibmobjectcsi.go:124]: Is this method getting used anywhere?

File: controllers/syncer/csi_node.go [hunk 3]
```diff
@@ -181,7 +185,7 @@ func (s *csiNodeSyncer) ensureContainer(name, image string, args []string) corev
 		Args:         args,
 		Env:          s.getEnvFor(name),
 		VolumeMounts: s.getVolumeMountsFor(name),
-		Resources:    ensureDefaultResources(),
+		//Resources:    ensureDefaultResources(),
```
Comment (ashimagarg27) [inline controllers/syncer/csi_node.go:188]: Please remove the commented code and I guess after this PR, method `ensureDefaultResources()` has become unused, please confirm and do cleanup accordingly

**Review by ashimagarg27 — CHANGES_REQUESTED** [review]
Changes looks good, added minor review suggestions, ptal.

**Review by nkkashyap — APPROVED** [review]
lgtm remove `ensureDefaultResources` as it is not used any more

**Review by ashimagarg27 — APPROVED** [review]
PR looks good!!

**Pull request merged**